pkg/karmadactl/cmdinit/kubernetes/statefulset.go

```diff
@@ -300,8 +299,8 @@ func (i *CommandInitOption) makeETCDStatefulSet() *appsv1.StatefulSet {
 		Volumes: *Volumes,
 	}
 
-	if i.EtcdStorageMode == "hostPath" && i.EtcdNodeSelectorLabels != "" {
-		podSpec.NodeSelector = utils.StringToMap(i.EtcdNodeSelectorLabels)
+	if i.EtcdStorageMode == "hostPath" && i.EtcdNodeSelectorLabelsMap != nil {
+		podSpec.NodeSelector = i.EtcdNodeSelectorLabelsMap
 	}
 	if i.EtcdStorageMode == "hostPath" && i.EtcdNodeSelectorLabels == "" {
 		podSpec.NodeSelector = map[string]string{"karmada.io/etcd": ""}
```

Comment:
How about update like this:
```go
if i.EtcdStorageMode == "hostPath" {
		if i.EtcdNodeSelectorLabelsMap != nil {
			podSpec.NodeSelector = i.EtcdNodeSelectorLabelsMap
		} else {
			podSpec.NodeSelector = map[string]string{"karmada.io/etcd": ""}
		}
	}
```

Reply:
This is actually better, I fixed the above problem, now how does the code look